pkg/cmd/pulumi/config.go

```diff
@@ -282,6 +282,8 @@ func newConfigSetCmd(stack *string) *cobra.Command {
 			switch {
 			case len(args) == 2:
 				value = args[1]
+			case !cmdutil.Interactive():
+				return errors.New("config value must be specified in non-interactive mode")
```

Comment:
This breaks the following:

```bash
$ pulumi config set foo < value.txt 
error: config value must be specified in non-interactive mode
```

The new check needs to be moved _after_ the `case !terminal.IsTerminal(int(os.Stdin.Fd())):` check.

Reply:
Fixed